### PR TITLE
Fix rpc get_events endpoint

### PR DESCRIPTION
**Pull request by THenry14**
## Usage related changes

Fixed some problems with rpc `get_events`:
- `from_block` and `to_block` are now accepted as block_hashes
- fixed `from_block` and `to_block` format in payload, so it now reflects the rpc spec
- fixed problems with response schema validation

There is still a probllem with (in my opinion) a bug in rpc schema regarding Felts (they are in `"^0x0[a-fA-F0-9]{1,63}$"` format, that is different to gateway), that's why I had to wrap test data with `rpc_felt`. From what I know, the discussion is already opened with starkware cairo team to resolve this. 

## Checklist:

- [X] Applied formatting - `./scripts/format.sh`
- [X] No linter errors - `./scripts/lint.sh`
- [X] Performed code self-review
- [X] Rebased to the last commit of the target branch (or merged it into my branch)
- [X] Documented the changes
- [X] Linked the issues which this PR resolves
- [X] Updated the tests
- [X] All tests are passing - `./scripts/test.sh`

### CODE REVIEW

**Comment by FabijanC** [comment]
@THenry14 Thanks for the PR! As you can see, the testing workflows have failed. I will probably review this next week.

**Comment by THenry14** [comment]
> @THenry14 Thanks for the PR! As you can see, the testing workflows have failed. I will probably review the PR next week after you have fixed this.

Oh, I'm sorry for that - I have run the linter, but then rebased and missed one unused import. I have addressed that in the newest commit, the tests should pass now :+1:

**Comment by FabijanC** [comment]
@THenry14
> I can see they failed on "Check out previous test metadata" step, not sure what it does and if it's connected to the code at all

Thanks for fixing the linter issue. What we have now is most likely a CircleCI issue, I will check it out.

**Review by FabijanC — CHANGES_REQUESTED** [review]
The CircleCI issue doesn't seem to be resolvable from my end. The downloading of previous test results (for splitting between workers) doesn't seem to be executed (file not present in a later check step). I tried writing a report on CircleCI's discussion forum only to get my post marked as spam for providing too many links :rofl: .

Anyway, thank you for raising the quality of Devnet's code, just left some minor comments.

File: starknet_devnet/blueprints/rpc/misc.py [hunk 1]
```diff
@@ -31,24 +34,34 @@ def check_address(address, event):
     """
     Check address.
     """
-    return bool(address is None or event.from_address == int(address, 0))
+    return bool(event.from_address == int(address, 0))
```
[the remark this answers is not in the document]
Comment (FabijanC) [reply starknet_devnet/blueprints/rpc/misc.py:37]: Actually here, what if the address is `None`? That's what has been checked so far. Or it cannot be `None`?

File: test/rpc/test_data/get_events.py [hunk 2]
```diff
@@ -1,114 +1,140 @@
...
 )
 
+from starknet_devnet.blueprints.rpc.utils import rpc_felt
+
 BLOCK_FROM_0_TO_LATEST = {
-    "from_block": "0",
+    "from_block": {"block_number": 0},
+    "address": rpc_felt(EXPECTED_FEE_TOKEN_ADDRESS),
+    "keys": [
+        rpc_felt("0x99cd8bde557814842a3121e8ddfd433a539b8c9f14bf31ebf108d12e6196e9"),
```
[the remark this answers is not in the document]
Comment (mikiw) [reply test/rpc/test_data/get_events.py:16]: `0x99cd8bde557814842a3121e8ddfd433a539b8c9f14bf31ebf108d12e6196e9` key was related with some contract that we use to test events. I'll take a look into that.

[the remark this answers is not in the document]
Comment (mikiw) [reply test/rpc/test_data/get_events.py:16]: Ok, so key is hex for a emited a event name.

[the remark this answers is not in the document]
Comment (mikiw) [reply test/rpc/test_data/get_events.py:16]: `0x3db3da4221c078e78bd987e54e1cc24570d89a7002cefa33e548d6c72c73f9d` is key for `increase_balance_called` event

[the remark this answers is not in the document]
Comment (mikiw) [reply test/rpc/test_data/get_events.py:16]: `0x99cd8bde557814842a3121e8ddfd433a539b8c9f14bf31ebf108d12e6196e9` is a key for some event related to fee estimation/charging (I'm not an expert here but I know that some fee estimations are generating event)

File: starknet_devnet/blueprints/rpc/misc.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (THenry14) [reply starknet_devnet/blueprints/rpc/misc.py]: My understanding is that it cannot be `None`, but I feel the spec is a bit vague when it goes for required fields, so there is a chance I'm mistaken. I think 0.11 should improve this situation.

File: starknet_devnet/blueprints/rpc/misc.py [hunk 3]
```diff
@@ -31,24 +34,34 @@ def check_address(address, event):
...
 
 
 def get_events_from_block(block, address, keys):
     """
     Return filtered events.
     """
     events = []
-    for event in [e for r in block.transaction_receipts for e in r.events]:
+    for transaction, event in [
+        (r, e) for r in block.transaction_receipts for e in r.events
```
[the remark this answers is not in the document]
Comment (THenry14) [reply starknet_devnet/blueprints/rpc/misc.py:53]: Oh, sorry. Thinking about it now, I have no idea why I have named it that way, perhaps I had some brainfart. Changed in newest commit.

File: test/rpc/test_data/get_events.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (THenry14) [reply test/rpc/test_data/get_events.py]: Sure, moved both events' keys to test.shared in newest commit.

**Review by FabijanC — APPROVED** [review]

**Pull request merged**